content/execution_model.tex

```diff
@@ -1,18 +1,18 @@
 An \openshmem program consists of a set of \openshmem processes called \acp{PE}
-that execute in a \ac{SPMD}-like model where each \ac{PE} can take a different
+that execute in a \ac{SPMD}-like model, where each \ac{PE} can take a different
 execution path. For example, a \ac{PE} can be implemented using an OS
 process.  The \acp{PE} progress asynchronously, and can communicate/synchronize
 via the \openshmem interfaces.  All \acp{PE} in an \openshmem program should
 start by calling the initialization routine  \FUNC{shmem\_init}
 \footnote{\textbf{start\_pes} has been deprecated as of Specification 1.2}
 before using any of the other \openshmem library routines.  An \openshmem
-program finishes execution by returning from the main routine or when any PE
+program finishes execution by returning from the main routine, or when any PE
 calls \FUNC{shmem\_global\_exit}. When returning from main, \openshmem must
 complete all pending communication and release all the resources associated to
 the library using an implicit collective synchronization across PEs. The user
 has the option to call \FUNC{shmem\_finalize} (before returning from main) to
-complete all pending communication and release all the \openshmem library
-resources without terminating the program. Calling any \openshmem routine after
+complete all pending communication, and release all the \openshmem library
```

Comment:
See _Comma Abuse No. 13_ in [Purdue's Extended Rules for Using Commas](https://owl.english.purdue.edu/owl/resource/607/02/).